lib/adminLogs.ts

```diff
@@ -0,0 +1,126 @@
+import dbConnector from "@lib/integration/dbConnector";
+import executeQuery from "@lib/integration/queryManager";
+import { logMessage } from "@lib/logger";
+
+enum Action {
+  Create = "Create",
+  Read = "Read",
+  Update = "Update",
+  Delete = "Delete",
+}
+
+/**
+ * Logs usage of privileged functions by admins
+ * @param adminSession
+ * @param action
+ * @param event
+ * @param description
+ */
+const logAdminActivity = async (
+  userId: string,
+  action: Action,
+  event: string,
+  description: string
+): Promise<void> => {
+  try {
+    await executeQuery(
+      await dbConnector(),
+      "INSERT INTO admin_logs (user_id, action, event, description) VALUES ($1, $2, $3, $4)",
+      [userId, action.toString(), event, description]
+    );
+  } catch (error) {
+    logMessage.error(error as Error);
+  }
+};
+
+export const logAdminRoleUpdate = async (
+  userAdminId: string,
+  isAdmin: boolean,
+  userId: string
+): Promise<void> => {
+  return logAdminActivity(
+    userAdminId,
+    Action.Update,
+    isAdmin ? "GrantAdminRole" : "RevokeAdminRole",
+    `Admin role has been ${isAdmin ? "granted" : "revoked"} for user id:${userId}`
+  );
+};
+
+export const logFormUpload = async (userAdminId: string, formId: string): Promise<void> => {
+  return logAdminActivity(
+    userAdminId,
+    Action.Create,
+    "UploadForm",
+    `Form (id:${formId}) has been uploaded`
+  );
+};
+
+export const logFormUpdate = async (userAdminId: string, formId: string): Promise<void> => {
+  return logAdminActivity(
+    userAdminId,
+    Action.Update,
+    "UpdateForm",
+    `Form (id:${formId}) has been updated`
+  );
+};
+
+export const logFormDelete = async (userAdminId: string, formId: string): Promise<void> => {
+  return logAdminActivity(
+    userAdminId,
+    Action.Delete,
+    "DeleteForm",
+    `Form (id:${formId}) has been deleted`
+  );
+};
+
+export const logFormBearerTokenRefresh = async (
+  userAdminId: string,
+  formId: string
+): Promise<void> => {
+  return logAdminActivity(
+    userAdminId,
+    Action.Update,
+    "RefreshBearerToken",
+    `Bearer token for form id:${formId} has been refreshed`
+  );
+};
+
+export const logFormAccessAddEmail = async (
+  userAdminId: string,
+  email: string,
+  formId: string
+): Promise<void> => {
+  return logAdminActivity(
+    userAdminId,
+    Action.Create,
+    "GrantInitialFormAccess",
+    `Email ${email} has been given access to form id:${formId}`
```

Comment:
Consistency in the format would be ideal, with colons and spacing.

`Email: ${email} has been given access to form id: ${formId}`